File: app/src/main/java/com/hbjpro/androidvoiceassistant/Command/ModuleApp.java
package com.hbjpro.androidvoiceassistant.Command;

import android.content.Intent;
import android.content.pm.ApplicationInfo;
import android.content.pm.PackageManager;

import com.hbjpro.androidvoiceassistant.Tools.GlobalState;

import java.util.Iterator;
import java.util.List;

public class ModuleApp {
    private PackageManager _packageManager;
    private List<ApplicationInfo> _listApp;

    public ModuleApp(){
        _packageManager = GlobalState.getInstance().getContext().getPackageManager();
        _listApp = _packageManager.getInstalledApplications(PackageManager.GET_META_DATA);
    }

    /* --- Public Methods --- */
    public Intent getAppLaunchIntent(String appName){
        Intent launchIntent = null;

        String packageName = getPackageName(appName);

        if(!packageName.isEmpty()){
            launchIntent = _packageManager.getLaunchIntentForPackage(packageName);
        }

        return  launchIntent;
    }

    /* --- Private Methods --- */
    private String getPackageName(String appName){
        Iterator it = _listApp.iterator();

        String packageName = "";

        while (it.hasNext()){
            ApplicationInfo app = (ApplicationInfo)it.next();
            String appLabel = _packageManager.getApplicationLabel(app).toString();

            if(appLabel.toLowerCase().equals(appName.toLowerCase())){
                packageName = app.packageName;
            }
        }

        return packageName;
    }
}

File: app/src/main/java/com/hbjpro/androidvoiceassistant/command/ModuleApp.kt
package com.hbjpro.androidvoiceassistant.command

import android.content.Intent
import android.content.pm.ApplicationInfo
import android.content.pm.PackageManager
import com.hbjpro.androidvoiceassistant.common.app.PackageManagerClient

class ModuleApp {

    /* --- Public Methods --- */
    fun getAppLaunchIntent(appName: String, callback: AppCallback<Intent>){
        var packageName = getPackageName(appName)

        if(!packageName.isEmpty()){
            callback.onSuccess(PackageManagerClient.getPackageManager().getLaunchIntentForPackage(packageName))
        }else{
            callback.onError("App could not be found")
        }
    }

    /* --- Private Methods --- */
    private fun getPackageName(appName: String): String{
        var app:ApplicationInfo? = PackageManagerClient.getPackageManager()
                .getInstalledApplications(PackageManager.GET_META_DATA)
                .find { PackageManagerClient.getPackageManager().getApplicationLabel(it).toString().toLowerCase() == appName }

        return if(app != null) app.packageName else ""
    }

    interface AppCallback<T>{
        fun onSuccess(result: T)
        fun onError(errorMsg: String)
    }
}